### ISSUE TITLE
`command not found: jest`

### ISSUE THREAD

**Issue by taschetto** [jest-community/vscode-jest#899]
### Environment

1. `vscode-jest version`: 4.6.0
2. `node -v`: 14.20
3. `npm -v` or `yarn --version`: 8.18.0
4. `npm ls jest` or `npm ls react-scripts` (if you haven’t ejected):

```
backend@0.1.0 /home/taschetto/projects/org/backend
├─┬ jest-watch-typeahead@1.1.0
│ └── jest@27.5.1 deduped
├── jest@27.5.1
└─┬ ts-jest@27.1.5
  └── jest@27.5.1 deduped
```
5. your vscode-jest settings if customized:

```
  "jest.autoRun": "off",
  "jest.jestCommandLine": "node_modules/.bin/jest", // same behavior when `undefined` or just `npm run tests`
  "jest.shell": "/bin/zsh" // same behavior when `undefined`
```

7. Operating system: Ubuntu 22.04 LTS

### Prerequisite
- are you able to run jest test from the command line? yes
- how do you run your tests from the command line? (for example: `npm run test` or `node_modules/.bin/jest`) `npm run test`

### Steps to Reproduce

We have a monorepo with the following structure:

```
.
├── jest.config.js
├── node_modules
├── package.json
├── package-lock.json
├── packages
└── test
```

The extension works fine for projects inside `packages/`. But we also have a project at the root level `.` (CDK project, tests inside `./tests`, can run tests in the terminal with `npm run tests`)

Whenever I try to run tests via the Text Explorer for that particular project, I get:

```
all-tests-56 is scheduled
zsh:1: command not found: jest
```

### Expected Behavior

It should run the tests.

### Actual Behavior

```
all-tests-56 is scheduled
zsh:1: command not found: jest
```

**Comment by taschetto** [comment]
Installing `jest` as a global package seems to solve the issue, but I don't believe it's the best solution.

**Comment by connectdotz** [comment]
This sounds like a monorepo project set-up issue. Please take a look at the [troubleshooting/jest-failed-to-run](https://github.com/jest-community/vscode-jest#jest-failed-to-run), particularly the [monorepo projects](https://github.com/jest-community/vscode-jest#how-to-use-the-extension-with-monorepo-projects) section. Let me know if it helps.

**Comment by gedclack** [comment]
Also happens to me after updated my VSCode to the newest (Version: 1.71.1 (Universal)).

I am using multi-root workspace.

Before updating my VSCode, all works fine, even the Test Editor shows correctly.

**Comment by taschetto** [comment]
> This sounds like a monorepo project set-up issue. Please take a look at the [troubleshooting/jest-failed-to-run](https://github.com/jest-community/vscode-jest#jest-failed-to-run), particularly the [monorepo projects](https://github.com/jest-community/vscode-jest#how-to-use-the-extension-with-monorepo-projects) section. Let me know if it helps.

The setup is correct.

**Comment by connectdotz** [comment]
@taschetto, I can take a deeper look if you can put together a sample repo.

@gedclack, does it return to normal if you restart `code`? I have seen the restart from the new vscode version sometimes caused shell env not to be initialized properly...

**Comment by gedclack** [comment]
@connectdotz Yes! I am on MacOS, I need to **quit** VSCode and open it again. It does not work if I only Close the editor window.

**Comment by javifm86** [comment]
Same here, but extension is not working, although I quit VS Code. I have a monorepo, everything was working fine (I configured some months ago).

I started getting this message around the end of August. Not only that, but I guess something has changed in VS Code, when I shut down my macOS (keeping my applications open for the next session), I get a warning message about some environment variable could not be initialized.

![image](https://user-images.githubusercontent.com/11178810/193250788-ad1d82db-ee13-47f3-9b00-bd4a1b463ef1.png)

The problem is related to NVM, but if I restart VS Code that warning disappears and shell environment are loaded successfully.

But this is still there:

![image](https://user-images.githubusercontent.com/11178810/193248624-6639f5fd-afb4-4ded-8f76-fd3c769eba69.png)

Just in case it can be related in any way. Maybe this information can help you, maybe not.

**Comment by connectdotz** [comment]
@javifm86 are you able to examine the process env of the spawned process? exit code 127 means "command not found", which usually indicates process.env is not fully set up as expected. See [self-diagnosis](https://github.com/jest-community/vscode-jest#how-to-see-more-debug-info-self-diagnosis) for how to check process.env. 

It is interesting that it was working fine until Aug...🧐

**Comment by javifm86** [comment]
Sure. Today the frontend part is working fine (I would swear I checked it on Fridays, and it was not working), but the backend part is not working. Here you have the relevant information from DevTools for backend:

```
[Extension Host] [backend/ProcessSessionManager] scheduling jest process: list-test-files
{
    "ELECTRON_RUN_AS_NODE": "1",
    "USER": "ambes000083",
    "MallocNanoZone": "0",
    "__CFBundleIdentifier": "com.microsoft.VSCode",
    "COMMAND_MODE": "unix2003",
    "LOGNAME": "ambes000083",
    "PATH": "/Users/ambes000083/.nvm/versions/node/v16.13.1/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin",
    "SSH_AUTH_SOCK": "/private/tmp/com.apple.launchd.H7HAitRYTk/Listeners",
    "SHELL": "/bin/zsh",
    "HOME": "/Users/ambes000083",
    "__CF_USER_TEXT_ENCODING": "0x1F5:0x0:0x0",
    "TMPDIR": "/var/folders/0g/2kdy6xf13ddcgg5n1bxkdmyc0000gn/T/",
    "XPC_SERVICE_NAME": "application.com.microsoft.VSCode.28571057.28571063",
    "XPC_FLAGS": "0x0",
    "ORIGINAL_XDG_CURRENT_DESKTOP": "undefined",
    "VSCODE_CWD": "/",
    "VSCODE_NLS_CONFIG": "{\"locale\":\"en\",\"availableLanguages\":{},\"_languagePackSupport\":true}",
    "VSCODE_CODE_CACHE_PATH": "/Users/ambes000083/Library/Application Support/Code/CachedData/74b1f979648cc44d385a2286793c226e611f59e7",
    "VSCODE_IPC_HOOK": "/Users/ambes000083/Library/Application Support/Code/1.71.2-main.sock",
    "VSCODE_PID": "3100",
    "SHLVL": "0",
    "PWD": "/",
    "OLDPWD": "/",
    "ZSH": "/Users/ambes000083/.oh-my-zsh",
    "PAGER": "less",
    "LESS": "-R",
    "LSCOLORS": "Gxfxcxdxbxegedabagacad",
    "NVM_DIR": "/Users/ambes000083/.nvm",
    "NVM_CD_FLAGS": "-q",
    "NVM_BIN": "/Users/ambes000083/.nvm/versions/node/v16.13.1/bin",
    "NVM_INC": "/Users/ambes000083/.nvm/versions/node/v16.13.1/include/node",
    "_": "/Applications/Visual Studio Code.app/Contents/MacOS/Electron",
    "VSCODE_AMD_ENTRYPOINT": "vs/workbench/api/node/extensionHostProcess",
    "VSCODE_HANDLES_UNCAUGHT_ERRORS": "true",
    "APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL": "true"
}
[Extension Host] [backend/ListTestFileListener] not-test onExecutableStdErr: /bin/sh: jest: command not found
notificationsAlerts.ts:42 not-test onProcessExit: process exit with code=127, signal=undefined
```

**Comment by connectdotz** [comment]
It seems vscode didn't initialize the process env completely. For people who frequently bumped into this issue, you can try switching to [login shell](https://github.com/jest-community/vscode-jest/blob/master/release-notes/release-note-v5.md#5-login-shell-support) added in v5 (still pre-release), and let us know if it resolved your issue.

**Comment by connectdotz** [comment]
@taschetto I looked at your original issue content more closely today, and noticed your comment:
```
 "jest.jestCommandLine": "node_modules/.bin/jest", // same behavior when `undefined` or just `npm run tests`
```
Not sure if it is a typo or a real problem, but you should set `"jest.jestCommandLine": "npm run test --"` to match how you run tests in the terminal. Your issue appeared to be different than the others in the thread, which is intermittent and most likely related to process.env. 

Anyway, let us know if that resolves your issue.

**Comment by connectdotz** [comment]
for people who landed here, we believe most of the issues have been addressed in [v5 pre-release](https://github.com/jest-community/vscode-jest/blob/master/release-notes/release-note-v5.md), please feel free to give it a spin and let us know otherwise.

**Comment by asifrahmanvm** [comment]
> login shell

Using the pre-release version v5.2 but still same problem. Sometimes it works other time it doesn't.

**Comment by connectdotz** [comment]
@asifrahmanvm, what is the error message when it fails?

**Comment by clxrityy** [comment]
I was encountering this error when trying to follow someone else's code, [this is the example repo](https://github.com/nickderaj/YouTube/blob/main/UI%20Library/package.json)

Every time I ran `pnpm i` and/or `pnpm update` I was also getting a separate error.

By doing `sudo pnpm i` then `sudo pnpm update`, my `pnpm test` (`jest watchAll --verbose`), worked fine. 

In short, I had to make sure the package was properly installed.
